eq-author-api/schema/tests/commentsIntroductionPage.test.js

```diff
@@ -0,0 +1,241 @@
+const { buildContext } = require("../../tests/utils/contextBuilder");
+
+const { createQuestionnaire } = require("../../utils/datastore");
+
+const {
+  deleteQuestionnaire,
+} = require("../../tests/utils/contextBuilder/questionnaire");
+const { BUSINESS } = require("../../constants/questionnaireTypes");
+
+const {
+  queryComments,
+  createComment,
+  deleteComment,
+  updateComment,
+  createReply,
+  updateReply,
+  deleteReply,
+} = require("../../tests/utils/contextBuilder/comments");
+
+describe("comments", () => {
+  let ctx, questionnaire, introductionId;
+
+  afterEach(async () => {
+    if (!questionnaire) {
+      return;
+    }
+    await deleteQuestionnaire(ctx, questionnaire.id);
+    questionnaire = null;
+  });
+
+  beforeEach(async () => {
+    ctx = await buildContext({
+      type: BUSINESS,
+      metadata: [{}],
+      sections: [
+        {
+          pages: [{}],
+        },
+        {},
+      ],
+    });
+
+    questionnaire = ctx.questionnaire;
+    introductionId = questionnaire.introduction.id;
+  });
+
+  it("An empty comment array is created on new questionnare", async () => {
+    const comment = await queryComments(ctx, {
+      introductionId: introductionId,
+    });
+
+    expect(comment.questionnaireIntroduction).toMatchObject({
+      id: expect.any(String),
+      comments: expect.any(Array),
+    });
+  });
+
+  it("should add a comment on questionnaireIntroduction page and then query that comment", async () => {
+    await createComment(ctx, {
+      introductionId: introductionId,
+      commentText: "a new comment is created",
+    });
+
+    const queryNewComments = await queryComments(ctx, {
+      introductionId: introductionId,
```

Comment:
You could reduce this object by doing the following

```suggestion
      introductionId,
```